pkg/runtime/runtime.go

```diff
@@ -626,6 +649,9 @@ func (m *MosnRuntime) initSecretStores(factorys ...*msecretstores.SecretStoresFa
 
 		// 2.3. save runtime related configs
 		m.secretStores[name] = comp
+		if err := m.initComponentInject(comp, config.ComponentRef); err != nil {
```

Comment:
How about injecting the components BEFORE comp.Init ? So that developers can use the component in the `Init` methods

![image](https://user-images.githubusercontent.com/26001097/192994697-bd416735-0b33-46e2-9c2b-a477c4836998.png)

Reply:
fixed